### ISSUE TITLE
Upgrade HTTP Sender to use to Apache HttpComponents

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#1956]
There are numerous advantages to updating to HttpComponents over HttpClient 3.x, mainly the documentation that is currently available.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-1985
Reporter: geraldb
Created: 2011-10-12T10:37:04.000-0700

**Comment by rbeckman-nextgen** [comment]
Upgraded to HttpComponents in revision 7174.

Imported Comment. Original Details:
Author: narupley
Created: 2014-05-27T14:04:51.000-0700

**Comment by rbeckman-nextgen** [comment]
Forgot to add a comment: Created a channel in 3.0.3 to use the http sender and all the available properties. This channel was then imported into 3.1 to verify that all the functionality still works properly. The new http sender now accepts the https scheme which is an upgrade from previous versions.

Imported Comment. Original Details:
Author: jaysenp
Created: 2014-08-20T16:55:41.000-0700

**Comment by rbeckman-nextgen** [comment]
Reopening as it looks like the 'Use Multipart' option is not including the correct Content-Type anymore. It is not clear if this ticket is what caused it but this was working as of 3.0.3. 

The top level Content-Type should have been sent at multipart/
`
<HttpRequest>
    <RemoteAddress>127.0.0.1</RemoteAddress>
    <RequestUrl>http://localhost:9440/</RequestUrl>
    <Method>POST</Method>
    <RequestPath/>
    <RequestContextPath>/</RequestContextPath>
    <Header>
        <Host>localhost:9440</Host>
        <Content-Length>328</Content-Length>
        <Accept-Encoding>gzip,deflate</Accept-Encoding>
        <User-Agent>Apache-HttpClient/4.3.3 (java 1.5)</User-Agent>
        <Connection>keep-alive</Connection>
        <Content-Type>text/x; charset=UTF-8</Content-Type>
    </Header>
    <Content multipart="no">--FKsBTasiWYK97V2UCUM1w0pMxh1VjiVcP&#13;
Content-Disposition: form-data; name="76cdab50-569c-4ced-85a7-f5d9022be2723890473554128789211.tmp"; filename="76cdab50-569c-4ced-85a7-f5d9022be2723890473554128789211.tmp"&#13;
Content-Type: text/x; charset=UTF-8&#13;
Content-Transfer-Encoding: binary&#13;
&#13;
test&#13;
--FKsBTasiWYK97V2UCUM1w0pMxh1VjiVcP--&#13;
</Content>
</HttpRequest>
`

Imported Comment. Original Details:
Author: eduardoa
Created: 2014-09-16T16:35:54.000-0700

**Comment by rbeckman-nextgen** [comment]
Fixed in revision 7381. The Content-Type header is no longer manually set when multipart is enabled. Also when the Content-Type specified in the connector settings doesn't have a charset defined but _does_ have other parameters, those other parameters are no longer wiped out by the dispatcher.

Imported Comment. Original Details:
Author: narupley
Created: 2014-09-17T16:13:46.000-0700

**Comment by rbeckman-nextgen** [comment]
This was already closed before I reopened it so I am reclosing it now that we verified these last two fixes. 

In rev 7372 if you sent the following in the Content Type field:
    plain/text;att1=one
Then then the listener would only see the following in the content type header:
    text/plain; charset=UTF-8

Confirmed that in rev 7384 the other attributes are no longer stripped. 

The other issue was that the content type was not being correctly set for the HTTP Sender multipart generated requests. The header should now include multipart/form-data and the boundary again. 

Imported Comment. Original Details:
Author: eduardoa
Created: 2014-09-19T08:34:56.000-0700
